### DISCUSSION TITLE
How to simulate a bigger step size for my GND analysis?

### DISCUSSION THREAD

**Discussion by matheusGariglio** [mtex-toolbox/mtex#924] [Ask Anything] [answered]
Hi :)

I'm currently trying to "simulate" a bigger step size in my EBSD data in order to evaluate the influence of the step size in my GND results.
For example, this ebsd data (1 is indexed data and 0 is not indexed):
1 1 1 1 
1 1 1 1 
1 1 1 1 
1 1 1 1
with a 2 times higher step size, would be like this:
1 0 1 0
0 0 0 0
1 0 1 0
0 0 0 0

I wrote the following code in order to pretend that my original ebsd (step size of 0.2 µm, for example) was acquired with a 2 times higher step size (ebsd2 step size = 0.4µm) :

```
Factor = 2; % how many times do you want to multiply your current step size
u = (ebsd(2).x - ebsd(1).x)*1e-6; %original step size

ebsd2 = ebsd;
a = 2:2:length(ebsd2);
ebsd2(ebsd.id(a)) = [] ; %remove half of the data

for i = 1 : length(ebsd2)
   uy = (ebsd(i).y - ebsd(1).y)*1e-6; %search where new line starts
   if uy > 0
      break
   end
end

j=(i-1)/Factor; % number of columns per ligne

for k = 1 : (length(ebsd)/Factor)/j-1 % remove all odd lines
   a = 1 + k*j : 1 : (k+1)*j;
   if a(end) <= length(ebsd2)
      ebsd2(ebsd.id(a)) = [] ;
   else
      break
   end
end

% for i = 1 : length(ebsd2)
%     ebsd2.id(i) = i;
% end

ux = (ebsd2(2).x - ebsd2(1).x)*1e-6; % check step size regarding x axis

for i = 1 : length(ebsd2)
   uy = (ebsd2(i).y - ebsd2(1).y)*1e-6; % check step size regarding y axis
   if uy > 0
      break
   end
end

% change unitcell to correspond to the new step size
ebsd2.unitCell = ebsd2.unitCell*Factor;

[ebsdGrid,newId] = gridify(ebsd2,'unitCell',ebsd2.unitCell);

figure; plot(ebsd2)

Factor = sprintf('%.0f',Factor);

if abs(ux - uy) <= 1e-9 || abs(ux - 2*u) <= 1e-9
   disp('Correct')
   ebsd2.export(strcat('1F_750°C_000_x1000_Stepx',Factor,'_EBSD_Filtered.ctf'))
else
   disp('Something is wrong with your data')
end
```
I would like to know if this is a correct way for simulating a higher step size in my ebsd data or should I do it in some other way.

Cheers,
Matheus

**Comment by ralfHielscher** [comment, marked as answer]
Have a look at the command EBSD/reduce

ralf.

**Reply by matheusGariglio** [reply]
Thanks, Ralf!

I didn't know this command.

Cheers,
Matheus

**Reply by matheusGariglio** [reply]
Hi Ralf,

When I do 
`ebsd2 = reduce(ebsd) `
and then try to calculate grains 
`[sousgrains,ebsd2.grainId] = calcGrains(ebsd2('indexed'),'angle',LAGB)`
I get the following error message:

```
Index exceeds the number of array elements (84525).

Error in spatialDecomposition (line 39)
  D = D(1:size(X,1));

Error in EBSD/calcGrains (line 45)
[V,F,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,varargin{:});

Error in Morphologie (line 177)
      [sousgrains,ebsd2.grainId] = calcGrains(ebsd2('indexed'),'angle',LAGB);
```
Is this some kind of bug?
You can find my original ebsd data in the following link : 

> https://cloud.mines-paristech.fr/index.php/s/gRQmbdITb897B8g

Cheers,
Matheus

**Reply by kilir** [reply]
Hi,

I can't reproduce your observation. Could you provide some more info?

Cheers,
Rüdiger

**Reply by matheusGariglio** [reply]
Hi Rüdiger,

Sure! Actually, I'm pretty sure that the error was coming from the fact that I exported the ebsd.
`ebsd2.export('New_EBSD.ctf')`

For information, I'm using MTEX 5.5.0.

If I do the following directly:
```
ebsd2 = reduce(ebsd);
[grains,ebsd2.grainId] = calcGrains(ebsd2('indexed'),'angle',10*degree);
```
Th grains are computed.
However, doing
```
ebsd2 = reduce(ebsd);
`ebsd2.export('New_EBSD.ctf')`
clear
import_wizard %add new ebsd2 variable from this exported data
[grains,ebsd2.grainId] = calcGrains(ebsd2('indexed'),'angle',10*degree);
```

I get the bug that I mentioned.

Cheers, 
Matheus

**Reply by kilir** [reply]
Using the current mtex version the following works:
```
ebsd = EBSD.load('1F_750°C_000_x1000_Original.ctf')
ebsd2 = reduce(ebsd)
ebsd2.export('reduced_EBSD.ctf')
%%
clear
ebsd = EBSD.load('reduced_EBSD.ctf')
[sousgrains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',1*degree)
```
There have been some bug fixes etc to `export_ctf` within the last year, so maybe you could try a more recent version.

Cheers,
Rüdiger

**Reply by matheusGariglio** [reply]
Ok, thanks for your answer. I'll check this out.

However, I downloaded the mtex 5.6.0 just after its release and I was having some troubles exporting my data with the `export_ctf` function, that's why I went back to the 5.5.0 version. I'll see if the 5.6.1 version works fine for me.

Cheers,
Matheus
